hs_discover/views.py

```diff
@@ -1,57 +1,111 @@
 import json
 
+from django.core.paginator import Paginator
 from django.shortcuts import render
 from django.views.generic import TemplateView
 from haystack.query import SearchQuerySet
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 
-# TODO error handling, validation, analytics
 class SearchView(TemplateView):
 
     def get(self, request, *args, **kwargs):
-        from django.template.defaultfilters import date, time
-        q = request.GET.get('q') if request.GET.get('q') else ""
+        return render(request, 'hs_discover/index.html', {})
 
-        sqs = SearchQuerySet()
 
-        vocab = []
-        for result in sqs:
-            if result.title:
-                vocab.extend(result.title.split(' '))
-            if result.subject:
-                vocab.extend(result.subject)
+class SearchAPI(APIView):
 
-        vocab = [x for x in vocab if len(x) > 2]
-        vocab = list(set(vocab))
-        vocab = sorted(vocab)
+    def get(self, request, *args, **kwargs):
```

Comment:
I thought drf-haystack was going to be used for serving discover.  I think if possible, we should use it instead of rolling our own paging and searching.  Also some of the code in this method, specifically the date formatting should probably be done in a serializer, not here.

Reply:
1) Regarding drf-haystack, I'll reach out to you in slack and post back here with a summary
2) Paging is a moving target and is still not fully settled. As such, custom code was needed to quickly iterate on changing requirements